packages/test-utils/src/index.js

```diff
@@ -0,0 +1,10 @@
+export { default as act } from './act';
```

Comment:
I must admit that it looks very familiar to the Puppeteer API. I did quick search in their documentation to find matching methods and nearly all of them are covered:
 - `act` – React testing helper
- `blur`
- `click` - https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pageclickselector-options
- `fire-event`
- `focus` – https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagefocusselector
- `hover` – https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagehoverselector
- `press` – https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-elementhandlepresskey-options
- `render` – React testing helper
- `type` – https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagetypeselector-text-options
- `wait` - https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagewaitforselectororfunctionortimeout-options-args

This makes my comment left on the parent issue even more tempting to explore:
https://github.com/WordPress/gutenberg/issues/17249#issuecomment-548462242